src/Cake/Scripting/Roslyn/RoslynCodeGenerator.cs

```diff
@@ -22,14 +22,37 @@ public string Generate(Script script)
 
         private static string GetAliasCode(Script context)
         {
-            var result = new List<string>();
+            var result = new Dictionary<string, string>();
             foreach (var alias in context.Aliases)
             {
-                result.Add(alias.Type == ScriptAliasType.Method
-                    ? MethodAliasGenerator.Generate(alias.Method)
-                    : PropertyAliasGenerator.Generate(alias.Method));
+                string hash, code = alias.Type == ScriptAliasType.Method
+                    ? MethodAliasGenerator.Generate(alias.Method, out hash)
+                    : PropertyAliasGenerator.Generate(alias.Method, out hash);
+
+                string @namespace = alias.Method.DeclaringType.Namespace ?? "@null";
+                if (result.ContainsKey(hash))
+                {
+                    var message = $"{alias.Type} \"{alias.Name}\" excluded from code generation and will need to be fully qualified on ICakeContext.";
```

Comment:
This is nitpicking but could we change this to

`The {alias.Type.ToLowerInvariant()} \"{alias.Name}\" was excluded from code generation and will need to be fully qualified on ICakeContext.`

Reply:
Addressed.